src/material/datepicker/src/datepicker.type.ts

```diff
@@ -18,7 +18,10 @@ import { MatDatepickerInput } from '@angular/material/datepicker';
       [placeholder]="to.placeholder"
       [tabindex]="to.tabindex || 0"
       [readonly]="to.readonly">
-    <ng-template #matSuffix>
+    <ng-template #matPrefix *ngIf="to.datepickerOptions.prefixIcon">
```

Comment:
what's about `datepickerTogglePosition` that allow the following values:
- suffix: `datepicker-toggle` as a suffix
- prefix: `datepicker-toggle` as a prefix
- null: without datepicker-toggle

Reply:
Sounds good to me. I'll update the PR.